internal/entities/models/models.go

```diff
@@ -52,3 +53,46 @@ func NewEntityWithProperties(dbEntity db.EntityInstance, props *properties.Prope
 		Properties: props,
 	}
 }
+
+// NewEntityWithPropertiesFromInstance creates a new EntityWithProperties instance from an existing entity instance
+func NewEntityWithPropertiesFromInstance(entity EntityInstance, props *properties.Properties) EntityWithProperties {
+	return EntityWithProperties{
+		Entity:     entity,
+		Properties: props,
+	}
+}
+
+// EntityForProperties gives us the necessary fields to fetch properties
+type EntityForProperties struct {
+	*EntityWithProperties
+
+	// Provider is the provider for the entity
+	Provider provifv1.Provider
```

Comment:
Why keep track of the provider and not just use the providerID in the EntityWithProperties? I wonder if the cleanest way would be to just pass the providerID to the propertyService, let the propertyService structure contain a providerManager and always instantiate the provider...

Reply:
Good observation. I was just trying to make it easier to instantiate everything. But this could actually be moved to the `RetrieveAllPropertiesForEntity` so we would not need the new wrapper. Let me refactor this.